designs/use-selectors-at-cache.md

```diff
@@ -0,0 +1,123 @@
+Filter cache ListWatch using selectors
+===================
+## Motivation
+
+The controller-runtime subscribe to changes using a cache mechanism that get
+updated using kubernetes informers, those informers implement a `ListWatch`
+that specify the namespace and type specified at controller builder time.
+
+So the only possible filtering of that cache is namespace and resource type,
+this means that if some user is interested only on Reconcile some resource with
+specific fields it will still cache all the instances on that type increasing
+the memory and CPU load. This also get worse in case controller-runtime is
+deployed as a daemonset instead of a deployment since all the nodes with
```

Comment:
I don't think how you deploy your controller is particularly relevant, the relevant aspect from your example is that your controller instance is only interested in a small subset of all objects that can not be filtered via namespace or type, right?

Reply:
Sure, removing it, your previous correction is good enough.